### DISCUSSION TITLE
Need help using noop datastore model read data from properties and returns standard json api response

### DISCUSSION THREAD

**Discussion by umasgowda** [yahoo/elide#2688] [Elide 5]
I have a model "DeliveryTerms" which does not interacts with any database, using noop datasource that configured in the application start up.

@SpringBootApplication
@EntityScan
public class APIServerApplication extends SpringBootServletInitializer {
    public static void main(String[] args) {
        SpringApplication.run(APIServerApplication .class, args);
    }

    @Override
    protected SpringApplicationBuilder configure(SpringApplicationBuilder application) {
        return application.sources(APIServerApplication .class);
    }

    @Bean
    public DataStore buildMultipleDataStore(EntityManagerFactory entityManagerFactory) {
        Consumer<EntityManager> txCancel = em -> em.unwrap(Session.class).cancelQuery();

        DataStore jpaStore = new JpaDataStore(
                entityManagerFactory::createEntityManager,
                em -> new NonJtaTransaction(em, txCancel)
        );

        DataStore noopStore = new CustomNoopDatastore(Collections.singletonList(DeliveryTerms.class));
        return new MultiplexManager(jpaStore, noopStore);
    }
}

Model example is: DeliveryTerms.java

@Include(name = "delivery-terms")
@Getter
@Setter
@LifeCycleHookBinding(operation = READ, phase = PRESECURITY, hook = DeliveryTermsService.class)
public class DeliveryTerms implements Serializable {
    @Id
    private String id;

    private String code;
    private String name;

}

LifeCycleHookBinding has wired with custome service class "DeliveryTermsService.class" which reads data from the properties and sets the DeliveryTerms
fields code and name.

DeliveryTermsService.java

@Component
public class DeliveryTermsService extends EmptyService<DeliveryTerms> {

    @Autowired
    @Qualifier("DeliveryTerms")
    private final String[] deliveryTermsFromProperty;

    public DeliveryTermsService(String[] deliveryTermsFromProperty) {
        this.deliveryTermsFromProperty = deliveryTermsFromProperty;
    }

    @Override
    public void reading(DeliveryTerms deliveryTerms, UserDetailsImpl userImp) {
        for (String deliveryTermStr : deliveryTermsFromProperty) {
            String[] deliveryTermsArray = deliveryTermStr.split(" - ");
            deliveryTerms.setCode(deliveryTermsArray[0]);
            deliveryTerms.setName(deliveryTermsArray[1]);
        }
    }
}

application.yml has 
deliveryterms: >
  ABC - I am at Place,
  DEF - Come to my Place,
  GHI - Where are you,
  KLM - Another property sample

------------------------------------

I would like to the json api response as follows: 

`
{
  "data": [
    {
      "type": "delivery-terms",
      "id": 1,
      "attributes": {
        "name" : "I am at Place",
        "code" : "ABC"
      }
    },
   {
      "type": "delivery-terms",
      "id": 2,
      "attributes": {
        "name" : "Come to my Place"
        "code" : "DEF"
      }
    }
   {...}
  ]
}
`
**Problem:** With the above code implementation using noop datastore it breaks the json api response and returns only one data element rather than collection of data elements. If I am going to implement JPA entity model, the JSON api response is as above. 

Can you please suggest me the solution to return the json api response similar to JPA entity models response? (json api response example is as above).

**Comment by umasgowda** [comment]
Use case is:

elide model should expose the data from the properties file ( application.yml file).
 Which data store would be good to implement when there is no database interaction ?

**Comment by aklish** [comment]
Your best bet is to create your own DataStore.  You can look at the [ConfigDataStore](https://github.com/yahoo/elide/blob/master/elide-model-config/src/main/java/com/yahoo/elide/modelconfig/store/ConfigDataStore.java) for inspiration.  It is similar in that in reads & writes to a configuration file.
